nixos/modules/virtualisation/cloudstack-config.nix

```diff
@@ -0,0 +1,43 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  imports = [
+    ../profiles/qemu-guest.nix
+  ];
+
+  config = {
+    fileSystems."/" = {
+      device = "/dev/disk/by-label/nixos";
+      autoResize = true;
+    };
+
+    boot.growPartition = true;
+    boot.kernelParams = [ "console=tty0" ];
+    boot.loader.grub.device = "/dev/vda";
+    boot.loader.timeout = 0;
+
+    # Allow root logins
+    services.openssh = {
+      enable = true;
+      permitRootLogin = "prohibit-password";
+    };
+
+    # Cloud-init configuration.
+    services.cloud-init.enable = true;
+    # Wget is needed for setting password. This is of little use as
+    # root password login is disabled above.
+    environment.systemPackages = [ pkgs.wget ];
+    # CloudStack data source is not robust enough to start without the
+    # network. Ensure dhcpcd.service has started.
+    systemd.services.cloud-init.after = [ "local-fs.target" "network.target" "cloud-init-local.service" "dhcpcd.service" ];
```

Comment:
As mentioned in #44524 we should fix this in the `cloud-init` module itself.

Reply:
Ack, I'll do that and update the PR.